Added Network support and ssh related packages

jayanthdc opened the pull request:
Intermingled logs on RPI3B+ https://github.com/OP-TEE/optee_client/issues/234

In this issue I wanted to ssh onto my rpi3 board so that the normal and secure world logs could be separated . During this course of action I realised my rpi3 board was not able to connect to the local network and to resolve it I have done the following.

I have added the below mentioned packages in order to get an IP assigned to the raspberry pi board and also to ssh onto it from the host Ubuntu. 
**Reason :** By default after buillding the image from the cloned the sources of the build.git for raspberrypi-3 (rpi3) board , I tried to ssh onto my board. 
I figured it out that these packages are missing and were needed to get an IP assigned from the dhcp and also ssh server related packages were needed inorder to connect to the rpi3 board from ssh client( Ubuntu).  
As it was clearly specified in the op-tee documentation that the default developer build contains minimal support and additional packages need to be added via Buildroot , I have added these packages here in common.mk file so that the buildroot takes care while building the Root file system. I have tested the same after adding these packages on my rpi3 board and was successful in getting an IP assigned from the dhcp and later ssh on to it.

**In this issue (https://github.com/OP-TEE/optee_client/issues/234) Intermingled logs on RPI3B+**  after discussing with @vchong I have sent this pull request. 
Hope this helps you in getting a clear picture of the issue and solution to it.
Thank you.


Internet Support related packages:
```
BR2_PACKAGE_DHCPCD ?= y
BR2_PACKAGE_ETHTOOL ?= y
BR2_PACKAGE_XINETD ?= y
```
SSH Packages :
```
BR2_PACKAGE_OPENSSH ?= y
BR2_PACKAGE_OPENSSH_SERVER ?= y
BR2_PACKAGE_OPENSSH_KEY_UTILS ?= y
```

Signed-off-by : Jayanth Dodderi Chidanand <jayanthchidanand94@gmail.com>
Reviewed-by : Joakim Bech <joakim.bech@linaro.org>
Acked-by : Etienne Carriere <etienne.carriere@linaro.org>

jbech-linaro commented:
Thanks for the patch, I'm in favor for enabling this for RPi3B+, however this causes a regression on the QEMU setup (as can be seen on the IBART logs). The `common.mk` is indeed common, so whatever you do there applies to other setups in build.git. So, I think you need to enable it from the `rpi3.mk` instead and do it before `include common.mk` line in that file.

jayanthdc commented:
> Thanks for the patch, I'm in favor for enabling this for RPi3B+, however this causes a regression on the QEMU setup (as can be seen on the IBART logs). The `common.mk` is indeed common, so whatever you do there applies to other setups in build.git. So, I think you need to enable it from the `rpi3.mk` instead and do it before `include common.mk` line in that file.

Thanks for your suggestion . I will do it accordingly.

etienne-lms reviewed (commented):
IBART does not complain while `common.mk` is still modified. No issue with IBART and Qemu?

As for the common part of it, i guess platforms willing these tools could dump the BR2 config.
`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>` with the `common.mk` change removed.
Despite my A-b, I wonder why a platform could not enable OpenSSH in Linux userland?

jbech-linaro commented:
> IBART does not complain while `common.mk` is still modified. No issue with IBART and Qemu?

That's weird on a previous patch it complained about BR2_PACKAGE_XINETD I think. But in any case, for now it's better to remove them from `common.mk` and just have them in the RPI makefile.

jayanthdc commented on rpi3.mk:
```diff
@@ -12,6 +12,16 @@ override COMPILE_S_KERNEL  := 64
 BR2_TARGET_GENERIC_GETTY_PORT ?= ttyS0
 BR2_ROOTFS_POST_BUILD_SCRIPT ?= "board/raspberrypi3-64/post-build.sh"
 
+#Network support related packages:
```
Ok.

jforissier commented:
> for now it's better to remove them from `common.mk`

This point is still TBD. Also, please squash all changes into a single commit and force push so we can see the final patch (there is no advantage in pushing incremental fixes to such a simple patch).
Thanks!

etienne-lms commented on common.mk:
```diff
@@ -253,6 +253,14 @@ BUILDROOT_TOOLCHAIN=toolchain-aarch$(COMPILE_NS_USER)-legacy
+#ssh packages
+BR2_PACKAGE_OPENSSH ?= y
+BR2_PACKAGE_OPENSSH_SERVER ?= y
+BR2_PACKAGE_OPENSSH_KEY_UTILS ?= y
```
remove this part of the change (keep only the change in rpi3.mk)

jayanthdc replied:
ok.

jayanthdc commented:
> > for now it's better to remove them from `common.mk`
> 
> This point is still TBD. Also, please squash all changes into a single commit and force push so we can see the final patch (there is no advantage in pushing incremental fixes to such a simple patch).
> Thanks!

Have done the things as insisted. Thanks for your comments.

jbech-linaro commented:
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

jforissier commented:
@jayanthdc I need a `Signed-off-by: Your RealName <your.email@some.domain>`, and please add Joakim's `Reviewed-by:` tag too. Please refer to older commits for examples. Thanks!

etienne-lms commented:
By the way, prefer update the commit log header line with a `rpi3: ` prefix,
i.e. "rpi3: enable network support and ssh related packages"

With comment addressed : `Acked-by: Etienne Carriere <etienne.carriere@linaro.org>`.

jayanthdc commented:
> @jayanthdc I need a `Signed-off-by: Your RealName <your.email@some.domain>`, and please add Joakim's `Reviewed-by:` tag too. Please refer to older commits for examples. Thanks!


Signed-off-by : Jayanth Dodderi Chidanand jayanthchidanand94@gmail.com
Reviewed-by : Joakim Bech joakim.bech@linaro.org
Acked-by : Etienne Carriere etienne.carriere@linaro.org

jayanthdc commented:
> > @jayanthdc I need a `Signed-off-by: Your RealName <your.email@some.domain>`, and please add Joakim's `Reviewed-by:` tag too. Please refer to older commits for examples. Thanks!
> 
> Signed-off-by : Jayanth Dodderi Chidanand [jayanthchidanand94@gmail.com](mailto:jayanthchidanand94@gmail.com)
> Reviewed-by : Joakim Bech [joakim.bech@linaro.org](mailto:joakim.bech@linaro.org)
> Acked-by : Etienne Carriere [etienne.carriere@linaro.org](mailto:etienne.carriere@linaro.org)

I have updated it in my main commit message as well. Please do correct me if something is wrong. Thank you.

jforissier commented:
> I have updated it in my main commit message as well. Please do correct me if something is wrong. Thank you.

Please update the commit message with `git commit --amend` and force push.

jayanthdc commented:
> > I have updated it in my main commit message as well. Please do correct me if something is wrong. Thank you.
> 
> Please update the commit message with `git commit --amend` and force push.

I have done it. Please check the the commit message I have changed and pushed it

jforissier commented:
Wrong format. Seriously @jayanthdc please look at past commits and do something correct.

jayanthdc commented:
> Wrong format. Seriously @jayanthdc please look at past commits and do something correct.

Sorry, doing it for the first time. I will do check it again and correct it.
